sdk/state/update_test.go

```diff
@@ -0,0 +1,75 @@
+package state
+
+import (
+	"testing"
+
+	"github.com/stellar/go/keypair"
+	"github.com/stellar/go/network"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestLastConfirmedPayment(t *testing.T) {
+	localSigner := keypair.MustRandom()
+	remoteSigner := keypair.MustRandom()
+	localEscrowAccount := &EscrowAccount{
+		Address:        keypair.MustRandom().FromAddress(),
+		SequenceNumber: int64(101),
+	}
+	remoteEscrowAccount := &EscrowAccount{
+		Address:        keypair.MustRandom().FromAddress(),
+		SequenceNumber: int64(202),
+	}
+	sendingChannel := NewChannel(Config{
+		NetworkPassphrase:   network.TestNetworkPassphrase,
+		Initiator:           true,
+		LocalSigner:         localSigner,
+		RemoteSigner:        remoteSigner.FromAddress(),
+		LocalEscrowAccount:  localEscrowAccount,
+		RemoteEscrowAccount: remoteEscrowAccount,
+	})
+	receiverChannel := NewChannel(Config{
+		NetworkPassphrase:   network.TestNetworkPassphrase,
+		Initiator:           false,
+		LocalSigner:         remoteSigner,
+		RemoteSigner:        localSigner.FromAddress(),
+		LocalEscrowAccount:  remoteEscrowAccount,
+		RemoteEscrowAccount: localEscrowAccount,
+	})
+
+	p, err := sendingChannel.ProposePayment(Amount{
+		Asset:  NativeAsset{},
+		Amount: 200,
+	})
+	require.NoError(t, err)
+
+	p, fullySigned, err := receiverChannel.ConfirmPayment(p)
+	assert.False(t, fullySigned)
+	require.NoError(t, err)
+	assert.Equal(t, p, receiverChannel.latestUnconfirmedPayment)
+
+	// Confirming a payment with same sequence number but different Amount should error
+	pDifferent := Payment{
+		IterationNumber: 1,
+		Amount: Amount{
+			Asset:  NativeAsset{},
+			Amount: 400,
+		},
+		CloseSignatures: p.CloseSignatures,
+	}
+	_, fullySigned, err = receiverChannel.ConfirmPayment(pDifferent)
+	assert.False(t, fullySigned)
+	require.Error(t, err)
+	require.Equal(t, "a different unconfirmed payment exists", err.Error())
```

Comment:
Should we also validate that the latestCloseAgreement remains empty?

Reply:
👍  https://github.com/stellar/experimental-payment-channels/pull/81/commits/23f87e013c43bc3fb906b7940915bb3ab9ceca85

also looks like [testify uses DeepEqual](https://github.com/stretchr/testify/blob/acba37e5db06f0093b465a7d47822bf13644b66c/assert/assertions.go#L58) for comparing structs, so asserting equal empty structs should be fine here